src/commands/up/app_source.rs

```diff
@@ -31,7 +32,13 @@ impl AppSource {
 
     pub fn infer_file_source(path: impl Into<PathBuf>) -> Self {
         match spin_common::paths::resolve_manifest_file_path(path.into()) {
-            Ok(file) => Self::File(file),
+            Ok(file) => {
+                if file.extension().and_then(std::ffi::OsStr::to_str) == Some("wasm") {
```

Comment:
Might be nice to support `*.wat` as well. If we want to be really precise we can use `wasmparser` to peek the special bytes to determine if the file is Wasm.

Reply:
> we can use `wasmparser` to peek the special bytes

...or `bytes.starts_with(b"\x00asm")` :slightly_smiling_face:

Reply:
This is probably the best thing to use: https://github.com/bytecodealliance/wasm-tools/blob/755abb691d9d401dad542b8f238d19e6f3624e16/crates/wat/src/lib.rs#L228

Reply:
> > we can use `wasmparser` to peek the special bytes
> 
> ...or `bytes.starts_with(b"\x00asm")` 🙂

...or https://github.com/bytecodealliance/wasm-tools/blob/755abb691d9d401dad542b8f238d19e6f3624e16/crates/wat/src/lib.rs#L228 😸

Reply:
I SUPPOSE

Reply:
This is absolutely world class bikeshedding, keep it up

Reply:
To add to the bikeshedding — do we actually expect users to try this with `.wat`?

Reply:
Most tools in this space accept either wat or wasm, so yes.

Reply:
I mean no, I don't think we expect users to use `.wat`, but also it isn't that hard to support apart from this slight inconvenience in detecting file type.

Reply:
For now I have checked for the `.wasm` and `.wat` extensions, and isolated the "is it a Wasm file" logic into a function that we can evolve with magic numbers and wasmparsers and other such wild excesses if and when people feel strongly about it.  And by "we" I mean you Brian